classes/local/renderer.php

```diff
@@ -66,41 +66,49 @@ public function get_data(): \stdClass {
     }
 
     /**
-     * Flattens an array with nested arrays into a single array.
+     * Loop through loops recursively to generate lists of variables.
      *
-     * @param array $output The output array to be written to.
-     * @param mixed $source The source array to be read from.
-     * @param string $prefix String to put at the front of the key name.
+     * @param object|array $source
+     * @param string $prefix
+     * @return array
      */
-    public static function flatten(array &$output, $source, string $prefix = '') {
+    private function looper($source, string $prefix = ''): array {
```

Comment:
So now we don't get a sample of the data that belong to each variable anymore just a list of the fields albeit in a nicer fashion?

Reply:
It's was proving too fiddly. You can a better representation from the preview.